taskcat/s3_stage.py

```diff
@@ -0,0 +1,149 @@
+
+class S3APIResponse:
+    def __init__(self, x):
+        self._http_code = x['ResponseMetadata']['HTTPStatusCode']
+
+    @property
+    def ok(self):
+        if self._http_code == 200:
+            return True
+        return False
+
+class S3BucketCreatorException(Exception):
+    pass
+
+class S3BucketCreator:
+    def __init__(self, config: Config):
+        self.name = ""
+        self.public = False
+        self.tags = []
+        self.region = 'us-east-1'
+        self.sigv4 = True
+        self._config = config
+        self._c = None
+
+    @property
+    def acl(self):
+        return self._acl
+
+    @property
+    def policy(self):
+        return self._policy
+
+    def _create_in_region(self, region):
+        if region == 'us-east-1':
+            response = self._c.create_bucket(
+                ACL=self.acl,
+                Bucket=self.name
+            )
+        else:
+            response = self._c.create_bucket(
+                ACL=self.acl,
+                Bucket=self.name,
+                CreateBucketConfiguration={
+                    'LocationConstraint': region
+                }
+            )
+
+        return S3APIResponse(response)
+
+    def _return_sigv4_policy(self, bucket):
+        policy = F"""{
```

Comment:
can this string go as a global or classvar, cleaner to have big blocks of text outside of the main code, also will ease localization if we ever get there.

 You'll have to drop the f-string, so the variable can be evaluated later:

POLICY = """blah {bucket} blah"""

policy = POLICY.format(bucket=my_bucket_name)